### PR TITLE
feat(cmd/server): add server executable

### LINKED ISSUE #18: server main
we should create an executable for the API server

### PR DESCRIPTION
**Pull request by KnutZuidema**
closes #18 
blocked by #17

### CODE REVIEW

File: cmd/server/main.go [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+	flag.StringVar(&ipAddress, "ip", "0.0.0.0", "ip address to listen on")
+	flag.IntVar(&port, "port", 5000, "port to listen on")
+	flag.StringVar(&dbAddress, "db", "postgres://postgres@postgres:5432?sslmode=disable",
+		"address of the postgres database")
+	flag.Parse()
+	ctx := context.Background()
+	logger := log.StandardLogger()
+	var db *sqlx.DB
+	var err error
+	for i := 0; i < 5; i++ {
```
Comment (frzifus) [inline cmd/server/main.go]: don't you think it would be pretty good to set the retries by flag?

Comment (KnutZuidema) [reply cmd/server/main.go]: sure

File: cmd/server/main.go [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+	flag.IntVar(&dbRetryWaitTime, "db.retrywait", 5, "time to wait between retries")
+	flag.Parse()
+	ctx := context.Background()
+	logger := log.StandardLogger()
+	var db *sqlx.DB
+	var err error
+	for i := 0; i < dbRetries; i++ {
+		db, err = sqlx.Connect("postgres", dbAddress)
+		if err != nil {
+			log.Info("postgres db is not ready, waiting 5 seconds:", err)
```
Comment (frzifus) [inline cmd/server/main.go]: shouldn't we take that time out of the flags?

**Review by frzifus — CHANGES_REQUESTED** [review]

File: cmd/server/main.go [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+	flag.IntVar(&dbRetryWaitTime, "db.retrywait", 5, "time to wait between retries")
+	flag.Parse()
+	ctx := context.Background()
+	logger := log.StandardLogger()
+	var db *sqlx.DB
+	var err error
+	for i := 0; i < dbRetries; i++ {
+		db, err = sqlx.Connect("postgres", dbAddress)
+		if err != nil {
+			log.Infof("postgres db is not ready, waiting %d seconds: %v", dbRetryWaitTime, err)
```
Comment (frzifus) [inline cmd/server/main.go]: lets use Warnf instead of Infof

Comment (KnutZuidema) [reply cmd/server/main.go]: of course master

**Review by frzifus — APPROVED** [review]

**Pull request merged**